src/aiokatcp/core.py

```diff
@@ -79,7 +79,10 @@ class Address:
     _IPV6_RE = re.compile(r"^\[(?P<host>[^]]+)\](:(?P<port>\d+))?$")
 
     def __init__(self, host: _IPAddress, port: Optional[int] = None) -> None:
-        self._host = host
+        if isinstance(host, typing.get_args(_IPAddress)):
+            self._host = host
+        else:
+            raise TypeError(f"{host} is not of either {typing.get_args(_IPAddress)}")
```

Comment:
Bikeshedding alert - feel free to ignore. I think this might be slightly more readable written as
```suggestion
        if not isinstance(host, typing.get_args(_IPAddress)):
            raise TypeError(f"{host} is not of either {typing.get_args(_IPAddress)}")
        self._host = host
```
Currently it looks like `self._host` is set conditionally while `self._port` is set unconditionally, but actually they're both set unless there is an exception.